spec/datasource_spec.js

```diff
@@ -171,4 +171,260 @@ describe('HawkularDatasource', function () {
+        end: options.range.to,
+        tags: "type:memory",
+        buckets: 1,
+        stacked: false
+      });
+
+      return ctx.$q.when({
+        status: 200,
+        data: [{
+          start: 13,
+          end: 19,
+          min: 15,
+          max: 25,
+          avg: 20.25
+        }]
+      });
+    };
+
+    ctx.ds.query(options).then(function (result) {
+      expect(result.data).to.have.length(1);
+      expect(result.data[0].datapoints).to.deep.equal([[20.25, 13]]);
+    }).then(v => done(), err => done(err));
+  });
+
+  it('should return live stats stacked', function (done) {
+
+    var options = {
+      range: {
+        from: 15,
+        to: 30
+      },
+      targets: [{
+        seriesAggFn: 'sum',
+        timeAggFn: 'live',
+        tags: [{name: 'type', value: 'memory'}],
+        type: 'gauge',
+        rate: false,
+        queryBy: 'tags'
+      }]
+    };
+
+    ctx.backendSrv.datasourceRequest = function (request) {
+      let pathElements = parsePathElements(request);
+      expect(pathElements).to.have.length(5);
+      expect(pathElements.slice(0, 2)).to.deep.equal(hPath.split('/'));
+      expect(pathElements.slice(2)).to.deep.equal(['gauges', 'raw', 'query']);
+      expect(request.data.limit).to.equal(1);
+      expect(request.data.tags).to.equal("type:memory");
+
+      return ctx.$q.when({
+        status: 200,
+        data: [{
+          id: "myhost.metric.memory.1",
+          data: [{
+            timestamp: 19,
+            value: 21
+          }]
+        },{
+          id: "myhost.metric.memory.2",
+          data: [{
+            timestamp: 19,
+            value: 25
+          }]
+        }]
+      });
+    };
+
+    ctx.ds.query(options).then(function (result) {
+      expect(result.data).to.have.length(1);
+      expect(result.data[0].datapoints).to.deep.equal([[46, 19]]);
+    }).then(v => done(), err => done(err));
+  });
+
+  it('should return live stats not stacked', function (done) {
+
+    var options = {
+      range: {
+        from: 15,
+        to: 30
+      },
+      targets: [{
+        seriesAggFn: 'avg',
+        timeAggFn: 'live',
+        tags: [{name: 'type', value: 'memory'}],
+        type: 'gauge',
+        rate: false,
+        queryBy: 'tags'
+      }]
+    };
+
+    ctx.backendSrv.datasourceRequest = function (request) {
+      let pathElements = parsePathElements(request);
+      expect(pathElements).to.have.length(5);
+      expect(pathElements.slice(0, 2)).to.deep.equal(hPath.split('/'));
+      expect(pathElements.slice(2)).to.deep.equal(['gauges', 'raw', 'query']);
+      expect(request.data.limit).to.equal(1);
+      expect(request.data.tags).to.equal("type:memory");
+
+      return ctx.$q.when({
+        status: 200,
+        data: [{
+          id: "myhost.metric.memory.1",
+          data: [{
+            timestamp: 19,
+            value: 21
+          }]
+        },{
+          id: "myhost.metric.memory.2",
+          data: [{
+            timestamp: 19,
```

Comment:
It has no incidence.

I had to take some arbitrary choices here about "live" aggregation: for instance, the timestamp returned to the grafana panel is the one of the first serie, in case there are multiple series. But it has no incidence because the singlestat panel doesn't use the timestamp for anything. And it's not displayed.

The other arbitrary decision I had was to consider for "live" metrics only metrics having a datapoint in the last 5 minutes. It may be subject to debate, but I can't just use the last available datapoint, because some old series may be like "terminated", like in the use case of openshift, when a container is shut down all associated series are still there, but they won't receive any new datapoint. So we need an arbitrary criteria to determinate what is "live".